src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -9,23 +8,45 @@ internal sealed class SpectralManager : ISpectralManager
 
     private readonly ILoggerWrapper _loggerWrapper;
     private readonly IHttpClientWrapper _httpClientWrapper;
-    private readonly string _toolDirectory;
-
-    public SpectralManager(LoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
+    private readonly string _spectralDirectory;
+    private readonly string _openApiToolsDirectory;
+    private readonly IProcessWrapper _processWrapper;
+    
+    public SpectralManager(ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper, IProcessWrapper processWrapper)
     {
         this._loggerWrapper = loggerWrapper;
         this._httpClientWrapper = httpClientWrapper;
-        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "spectral", SpectralVersion);
+        this._openApiToolsDirectory = openApiToolsDirectoryPath;
+        this._spectralDirectory = Path.Combine(openApiToolsDirectoryPath, "spectral", SpectralVersion);
+        this._processWrapper = processWrapper;
     }
+    
+    private string ExecutablePath { get; set; } = string.Empty;
 
     public async Task InstallSpectralAsync(CancellationToken cancellationToken)
     {
-        Directory.CreateDirectory(this._toolDirectory);
+        Directory.CreateDirectory(this._spectralDirectory);
 
-        var executableFileName = GetSpectralFileName();
-        var url = $"https://github.com/stoplightio/spectral/releases/download/v{SpectralVersion}/{executableFileName}";
+        this.ExecutablePath = GetSpectralFileName();
+        var url = $"https://github.com/stoplightio/spectral/releases/download/v{SpectralVersion}/{this.ExecutablePath}";
 
-        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._toolDirectory, executableFileName), cancellationToken);
+        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._spectralDirectory, this.ExecutablePath), cancellationToken);
```

Comment:
Could we consider the case where Spectral is already downloaded to save time?

Also if the download fails, a partial file might be already present. I believe `DownloadFileToDestinationAsync` should attempt to delete the downloaded file if an HTTP error happened.

Reply:
Since we are deciding which "version" to download. Does it make sense to just do a simple check with `Content.Length`? It won't be 100% accurate but it will do the work for now. Thoughts?